docs/RFCS/20190906_flow_control.md

```diff
@@ -0,0 +1,137 @@
+
+### Congestion definition
+
+There exists some key concepts in a data import/replication link:
+
+- Data loss: Each database transaction execution failure is treated as one data loss, the failure includes every kind of database execution failure that no data is written to downstream, partial data written error does not include.
+- Bandwidth: Bandwidth describes the maximum data import/replication rate from a DM-worker to downstream, this can be measured by TPS from DM-worker (or TPS in downstream, but should only include the data traffic from this DM-worker).
+- Latency: Latency is the amount of time it takes for data to replicate from DM-worker to downstream.
+
+The congestion usually means the quality of service decreases because of the service node or data link is carrying more data than it can handle. In the data import/replication scenario, the consequence of congestion can be partial downstream database execution timeout, downstream qps decrease or SQL execution latency increase. We can use these three indicators to determine whether congestion happens and measure the degree of congestion.
+
+#### How to measure the key indicators
+
+Before we can measure the congestion, we should find a way to estimate the data loss, the bandwidth and the latency.
+
+- Data loss: can be collected in DM-worker itself, and we often use it as data loss rate, which means the count of sqls in all failed transaction divided by the count sqls in all transactions that need to be processed.
+- Latency: equals to the transaction executed latency, can be collected in DM-worker itself.
+- Bandwidth: this is a little complicated, but we can estimate it by adjusting the concurrency and find the maximum downstream TPS for this DM-worker, which can be used as the bandwidth of the data import/replication link.
+
+Take the [benchmark result in DM 1.0-GA](https://pingcap.com/docs/v3.0/benchmark/dm-v1.0-ga/#benchmark-result-with-different-pool-size-in-load-unit) as an example, we extract the result in load unit test, use load unit pool size as X-axis, latency and load speed as Y-axis.
+
+![DM benchmark result with different pool size in load unit](../media/rfc-load-benchmark.png)
+
+From the benchmark result we can draw the following conclusions:
+
+- The latency grows exponentially as the concurrency increases.
+- The TPS grows linely as the concurrency increases in a specific range, in this case the liner range is between 0-16.
+- After the TPS reaches the maximum value, it remains the same even the concurrency increases. The maximum value of TPS can be used as the bandwidth.
+
+#### How to detect and measure congestion
+
+Congestion can be exposed in different ways and have different effect on key indicators. We have the following predictive indicators change that may reflect congestion.
+
+- If TPS doesn't increase alongside with the concurrency increase (assuming the dumped data or relay log stream is fast enough), we can believe the congestion is occurring.
```

Comment:
In short
- TPS does not increases and decreases
-  The transaction failure rate increases.